zkevm-circuits/src/keccak_circuit/keccak_packed_multi.rs

```diff
@@ -12,12 +12,43 @@ use std::{env::var, vec};
 
 const MAX_DEGREE: usize = 9;
 
-pub(crate) fn get_num_rows_per_round() -> usize {
+/// Obtain the rows required for 1 iteration of f-box's inner round
+/// function (consisting of 5 phases) within Keccak circuit
+pub fn get_num_rows_per_round() -> usize {
     var("KECCAK_ROWS")
         .unwrap_or_else(|_| format!("{DEFAULT_KECCAK_ROWS}"))
         .parse()
         .expect("Cannot parse KECCAK_ROWS env var as usize")
 }
+/// Obtain the rows required for 1 iteration of the f-box
+/// function (consisting of nr = 12 + 2*l inner rounds)
+/// within Keccak circuit
+pub fn get_num_rows_per_update() -> usize {
+    get_num_rows_per_round() * (NUM_ROUNDS + 1)
+}
+/// Obtain the column position of the hash inputs
+/// within cell_manager for an inner round.
+/// This value is determined by the number of rows allocated
+/// to each inner round and target part_size for u64
+pub fn get_input_bytes_col_cell_manager() -> usize {
+    let mut col: usize = 0;
+    let inner_round_num_rows = get_num_rows_per_round();
+
+    col += 28 / inner_round_num_rows;
+    if inner_round_num_rows * col < 28 {
+        col += 1;
+    }
+
+    let part_size = get_num_bits_per_absorb_lookup();
+    let part_length = WordParts::new(part_size, 0, false).parts.len();
+
+    let mut absorb_parts_col = part_length / inner_round_num_rows;
+    if inner_round_num_rows * absorb_parts_col < part_length {
+        absorb_parts_col += 1;
+    }
+
+    col + absorb_parts_col * 2 + 1
+}
```

Comment:
i would suggest (1) make 28 a named constant and add some explanation for it (2) add a test for get_input_bytes_col_cell_manager? With time goes by, without a test this function may goes wrong with unintended updates elsewhere.